src/shared/awsagent/roles.go

```diff
@@ -98,23 +103,42 @@ func (a *Agent) DeleteOtterizeIAMRole(ctx context.Context, accountName string) e
 	}
 
 	_, found := lo.Find(role.Tags, func(tag types.Tag) bool {
-		return *tag.Key == serviceAccountNameTagKey
+		if *tag.Key == serviceAccountNameTagKey && *tag.Value == accountName {
+			return true
+		}
+		if *tag.Key == serviceAccountNamespaceTagKey && *tag.Value == namespaceName {
+			return true
+		}
+
+		return false
 	})
 
 	if !found {
-		errorMessage := fmt.Sprintf("refusing to delete role that's not tagged with %s", serviceAccountNameTagKey)
-		logger.Error(errorMessage)
+		errorMessage := "refusing to delete role that's not tagged with with appropriate tags"
+		logger.WithField("arn", role.Arn).Error(errorMessage)
 		return errors.New(errorMessage)
 	}
 
-	_, err = a.iamClient.DeleteRolePolicy(ctx, &iam.DeleteRolePolicyInput{
+	_, err = a.iamClient.GetRolePolicy(ctx, &iam.GetRolePolicyInput{
 		PolicyName: role.RoleName,
 		RoleName:   role.RoleName,
 	})
 
-	if err != nil {
-		logger.WithError(err).Errorf("failed to delete inline policy")
-		return err
+	if err == nil {
+		_, err = a.iamClient.DeleteRolePolicy(ctx, &iam.DeleteRolePolicyInput{
+			PolicyName: role.RoleName,
+			RoleName:   role.RoleName,
+		})
+
+		if err != nil {
+			logger.WithError(err).Errorf("failed to delete inline policy")
+			return err
```

Comment:
If an error is returned, then don't log - let the logger above you handle logging (in this case it's controller-runtime's logger). If you want to add a custom message, use `fmt.Errorf("my custom message for error: %w", err)`, which will add a custom string to the error and also wrap the message so `errors.Is` and `errors.As` still works.